presto-main/src/main/java/com/facebook/presto/split/RemoteSplit.java

```diff
@@ -29,11 +30,13 @@
         implements ConnectorSplit
 {
     private final URI location;
+    private final TaskId sourceTaskId;
 
     @JsonCreator
-    public RemoteSplit(@JsonProperty("location") URI location)
+    public RemoteSplit(@JsonProperty("location") URI location, @JsonProperty("taskId") TaskId sourceTaskId)
```

Comment:
nit: Why don't we use the same name for both the parameter and the JSON property, e.g., either `taskId` for both or `sourceTaskId` for both?